sdk/resource/builtin.go

```diff
@@ -46,6 +46,10 @@ type (
 	}
 
 	defaultServiceNameDetector struct{}
+
+	// NoOp is a Detector that only provides an empty resource.  Used
```

Comment:
Perhaps consider the name "Empty" instead of "NoOp" here, though if "Empty" came after some other `Detector`s, it might sound as though it's going to clear the set and start again from nothing for any `Detector`s that follow.

_[Time passes ...]_

Oh, but then reading the documentation at line 109 below, it sounds like this _does_ clear any preceding `Resource`s. Or, a different reading, perhaps it offers an empty `Resource` to be merged, which would still count as a "no-op." I'll need to keep reading further to understand how we use the output of each `Detector`.

Reply:
Point taken, I did think of Empty but came to the same conclusion.  Because the detectors are overlays, later detectors will overwrite keys from earlier ones, I decided not to have any name that would imply that it erases any previous detector work.

If this attempt at cleaning up the API around resources is more useable we can have a discussion about what to call the `Detector` that doesn't detect anything before the merge.